### PR TITLE
Skeletonize

### LINKED ISSUE #2: Update this Ansible role
# 🐛 Summary #

This Ansible role is horrifically out of date:
- It's still wanting to use TravisCI
- The MongoDB sources should use an HTTPS endpoint instead of HTTP.

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request skeletonizes this repository against [cisagov/skeleton-ansible-role](https://github.com/cisagov/skeleton-ansible-role).

## 💭 Motivation and context ##

Resolves #2.

## 🧪 Testing ##

All automated testing passes.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Comment by jsf9k** [comment]
The failing test is an old LGTM Javascript test that doesn't even apply to this repo (since it contains no JS code).

File: .github/lineage.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+lineage:
+  skeleton:
+    remote-url: https://github.com/cisagov/skeleton-generic.git
```
Comment (dav3r) [inline .github/lineage.yml]: Shouldn't this role descend from `skeleton-ansible-role`?
```suggestion
    remote-url: https://github.com/cisagov/skeleton-ansible-role.git
```

Comment (jsf9k) [reply .github/lineage.yml]: Great catch!

File: meta/main.yml [hunk 2]
```diff
@@ -3,15 +3,39 @@ galaxy_info:
...
+    #   versions:
+    #     - 2
     - name: Debian
       versions:
         - stretch
-  galaxy_tags:
-    - mongodb
-    - mongo
+    #     - buster
+    #     - bullseye
```
Comment (dav3r) [inline meta/main.yml:25]: Is there a reason we aren't including buster and bullseye now?  I also don't remember why we excluded the other platforms here as well, but I'm less concerned about them.

Comment (jsf9k) [reply meta/main.yml:25]: This Ansible role [only ever supported Stretch](../blob/e8e178464576e7b54b51e9047eb2d9acaed0dfb5/files/mongodb-org-3.6.list#L1).  It probably does need to be expanded to support Buster (at a minimum) but I think that should be a separate PR.

Comment (dav3r) [reply meta/main.yml:25]: That works for me.  Can you please create an issue to support Buster (and beyond?) so we don't forget about it?

**Review by dav3r — APPROVED** [review]
Looks great, strong work, May the 4th be with you. ⭐

File: meta/main.yml (hunk 2 above)
Comment (jsf9k) [reply meta/main.yml:25]: Please see #4.

Comment (dav3r) [reply meta/main.yml:25]: Thank you sir!

File: tasks/main.yml [hunk 3]
```diff
@@ -1,36 +1,45 @@
...
 
 - name: Update the cache with the mongodb-org goodness
-  package:
+  ansible.builtin.package:
     update_cache: yes
 
 - name: Install mongo
-  package:
+  ansible.builtin.package:
     name: mongodb-org=3.6.8
```
Comment (mcdonnnj) [inline tasks/main.yml]: Should we just bump this to the latest point release?

```suggestion
    name: mongodb-org=3.6.23
```

File: tasks/main.yml [hunk 4]
```diff
@@ -78,61 +87,61 @@
...
 # status mongod.service".
 - name: Test for AssertPathIsMountPoint lines
-  command: grep -c ^AssertPathIsMountPoint= /lib/systemd/system/mongod.service
-  register: test_assertpathismountpoint
-  ignore_errors: yes
+  ansible.builtin.command: grep -c ^AssertPathIsMountPoint= /lib/systemd/system/mongod.service
   changed_when: false
+  ignore_errors: yes
+  register: test_assertpathismountpoint
 - name: Make sure mongo data file system is mounted before starting
```
Comment (mcdonnnj) [inline tasks/main.yml]: Should these three tasks that all have a `when: test_assertpathismountpoint.rc != 0` be moved into a `block`?

File: tasks/main.yml [hunk 5]
```diff
@@ -78,61 +87,61 @@
...
 # directory exists and has the correct permissions.
 - name: Test for RuntimeDirectory line
-  command: grep -c ^RuntimeDirectory= /lib/systemd/system/mongod.service
-  register: test_runtimedirectory
-  ignore_errors: yes
+  ansible.builtin.command: grep -c ^RuntimeDirectory= /lib/systemd/system/mongod.service
   changed_when: false
+  ignore_errors: yes
+  register: test_runtimedirectory
 - name: Make sure /run/mongodb is created before starting
```
Comment (mcdonnnj) [inline tasks/main.yml:124]: Same question about using a `block` for these `when: test_runtimedirectory.rc != 0` tasks.

File: tasks/main.yml [hunk 6]
```diff
@@ -78,61 +87,61 @@
...
-  lineinfile:
+  ansible.builtin.lineinfile:
     dest: /lib/systemd/system/mongod.service
     insertafter: ^RuntimeDirectory=mongodb
-    state: present
     line: RuntimeDirectoryMode=0744
+    state: present
   when: test_runtimedirectorymode.rc != 0
 
 # pymongo is needed for ansible mongodb_user Ansible module
```
Comment (mcdonnnj) [inline tasks/main.yml]:
```suggestion
# pymongo is needed for the mongodb_user Ansible module
```

**Review by mcdonnnj — APPROVED** [review]
Strong work 💪💪💪 I had some suggestions that are at your discretion but it's nice to see another repo brought into the light.

File: tasks/main.yml (hunk 4 above)
Comment (jsf9k) [reply tasks/main.yml]: Please see commit 491a4a1.

File: tasks/main.yml (hunk 5 above)
Comment (jsf9k) [reply tasks/main.yml:124]: The `when` clauses for those two tasks are different, are they not?

Comment (mcdonnnj) [reply tasks/main.yml:124]: Yep, missed the `mode` at the end of the second test.

**Pull request merged**